lstchain/reco/dl0_to_dl1.py

```diff
@@ -92,19 +94,33 @@ def get_dl1(calibrated_event, telescope_id, dl1_container=None, custom_config={}
 
     signal_pixels = cleaning_method(camera, image, **cleaning_parameters)
 
+
     if image[signal_pixels].sum() > 0:
-        hillas = hillas_parameters(camera[signal_pixels], image[signal_pixels])
+
+        # check the number of islands 
+        num_islands, island_labels = number_of_islands(camera, signal_pixels)
+        intensity_of_island = np.zeros(num_islands+1)
+
+        for iisland in range(1, num_islands+1):
+            intensity_of_island[iisland] = np.sum(image[island_labels==iisland])
+
+        max_island_label = np.argmax(intensity_of_island)
```

Comment:
I am wondering if the sum of intensity is the best approach here rather than the number of pixels - or a combination of both, the mean.
If we have two pixels with high signal, is it more likely to be the shower compared to 4-5 pixels with average signal (but passed the cleaning of course).

Reply:
I see. I modified the code to use # of pixels on island to search main island following your comment for the moment. I will study which way is the best later.

Reply:
It was a suggestion, I have no idea which could be better.
But if you can have a look that'd be great.
In the meantime, I think it's a good implementation.